### ISSUE TITLE
Error Message: React does not recognize the `dirtyFieldsSinceLastSubmit` prop on a DOM element.

### ISSUE THREAD

**Issue by sharon-xz** [marmelab/react-admin#3960]
I just updated from v2.9 to v3.0.0-beta.6. I followed the upgrade guide and everything works fine except the following error message:

Warning: React does not recognize the `dirtyFieldsSinceLastSubmit` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `dirtyfieldssincelastsubmit` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

The whole project can still work since it's still the early stage of my project. I hope this error can be fixed! Thanks!

**Related code:**

```
// at PlanList.js:69
<List {...props} bulkActionButtons={false} filters={<PlanFilter />} actions={<PlanActions />}>
...
</List>
```

```
// at  PlanFilter.js:
export default (props) => (
    <Filter variant="standard" {...props}>
        <SearchInput source="q" alwaysOn />
    </Filter>
);
```

**Environment**

* React-admin version: v3.0.0-beta.6
* Last version that did not exhibit the issue (if applicable): 2.9.8
* React version: 16.11.0
* Browser: Chrome
* Stack trace (in case of a JS error):

Warning: React does not recognize the `dirtyFieldsSinceLastSubmit` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `dirtyfieldssincelastsubmit` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in form (created by FilterForm)
    in FilterForm (created by ReactFinalForm)
    in ReactFinalForm (created by EnhancedFilterForm)
    in EnhancedFilterForm (created by Filter)
    in Filter (at PlanFilter.js:6)
    in Unknown (at PlanList.js:69)
    in div (created by ForwardRef(Toolbar))
    in ForwardRef(Toolbar) (created by WithStyles(ForwardRef(Toolbar)))
    in WithStyles(ForwardRef(Toolbar)) (created by ListToolbar)
    in ListToolbar (created by ListView)
    in div (created by ListView)
    in ListView (created by List)
    in List (at PlanList.js:69)
    in Unknown (created by WithPermissions)
    in WithPermissions (created by Context.Consumer)
    in Route (created by ResourceRoutes)
    in Switch (created by ResourceRoutes)
    in ResourceRoutes (created by Resource)
    in Resource (at App.js:22)
    in Route (created by RoutesWithLayout)
    in Switch (created by RoutesWithLayout)
    in RoutesWithLayout (created by Context.Consumer)
    in div (created by Layout)
    in main (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in Layout (created by WithStyles(Layout))
    in WithStyles(Layout) (created by Context.Consumer)
    in withRouter(WithStyles(Layout)) (created by ConnectFunction)
    in ConnectFunction (created by LayoutWithTheme)
    in ThemeProvider (created by LayoutWithTheme)
    in LayoutWithTheme (at Layout/index.js:5)
    in MyLayout (created by Context.Consumer)
    in Route (created by CoreAdminRouter)
    in Switch (created by CoreAdminRouter)
    in div (created by CoreAdminRouter)
    in CoreAdminRouter (created by Context.Consumer)
    in Route (created by AdminUI)
    in Switch (created by AdminUI)
    in AdminUI (created by Admin)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (created by AdminContext)
    in TranslationProvider (created by AdminContext)
    in Provider (created by AdminContext)
    in AdminContext (created by Admin)
    in Admin (at App.js:14)
    in App (at src/index.js:7)

**Comment by MiMo42** [comment]
see #3959

**Comment by djhi** [comment]
Fixed by #3959
